### PR TITLE
MB-1581 add zip sit address for sit destination

### PR DESCRIPTION
**Pull request by gidjin**
## Description

This PR adds the implementation for the `ZipSITDestHHGFinalAddress` lookup. It also renames the old `ZipSITAddress` to `ZipSITDestHHGFinalAddress` name for clarity, and updates the description for it and `ZipDestAddress`. 

## Reviewer Notes

***NOTE*** *This PR Depends on #5672 and should be merged after that one is.* So please only review the latest commits by me.

## Setup

```sh
make server_test
```

## Code Review Verification Steps

* Any new migrations/schema changes:
  * [X] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [X] Have been communicated to #g-database
* [X] Request review from a member of a different team.
* [X] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1581) for this change
* [Jira MB-6228](https://dp3.atlassian.net/browse/MB-6228)
* [Jira MB-6232](https://dp3.atlassian.net/browse/MB-6232)
* [Jira MB-6233](https://dp3.atlassian.net/browse/MB-6233)
* [Jira MB-6386](https://dp3.atlassian.net/browse/MB-6386)
* [Jira MB-6387](https://dp3.atlassian.net/browse/MB-6387)
* [this slack message](https://ustcdp3.slack.com/archives/CP6F568DC/p1611168514100800) explains more about the approach used.

### CODE REVIEW

**Review by pearl-truss — APPROVED** [review]
Your specific commits look good to me

File: migrations/app/schema/20210120194932_update_zip_sit_dest_hhg_final_address_service_param_key.up.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-- Rename ZipSITAddress to ZipSITDestHHGFinalAddress and update description
+UPDATE service_item_param_keys
+  SET key = 'ZipSITDestHHGFinalAddress', description = 'SIT Final Destination Address ZIP from MTOServiceItem'
+  WHERE key = 'ZipSITAddress';
```
Comment (reggieriser) [inline migrations/app/schema/20210120194932_update_zip_sit_dest_hhg_final_address_service_param_key.up.sql:4]: Nit: We could set the `updated_at` column to the current timestamp in these two UPDATE statements.

**Review by reggieriser — APPROVED** [review]
Reviewing on behalf of the database reviewers group.  The migration you added LGTM (with one small nit that's not a blocker).  I reviewed the other migration you brought in from #5672 in that PR.

File: migrations/app/schema/20210120194932_update_zip_sit_dest_hhg_final_address_service_param_key.up.sql (hunk 1 above)
Comment (gidjin) [reply migrations/app/schema/20210120194932_update_zip_sit_dest_hhg_final_address_service_param_key.up.sql]: Good call, fixed.

**Pull request merged**